tensorflow/compiler/tf2tensorrt/convert/utils.h

```diff
@@ -128,67 +129,223 @@ inline bool IsTrtShapeTensorCompatible(const Tensor& tensor) {
          IsTrtShapeTensorCompatible(tensor.shape());
 }
 
-template <typename Container>
-Status ContainerToTrtDims(const Container& shape, nvinfer1::Dims* trt_dims,
-                          bool ignore_first_dim = false) {
-  if (shape.size() == 0) {
-    // scalar
-    if (ignore_first_dim) {
-      return errors::Internal(
-          "Scalars cannot be represented in implicit batch mode");
+// Adapts various representations of shape (TF Shape, TRT Dims, plain
+// containers) and provides methods for properties (length, volume) and
+// conversion between types. Note that unlike TF's TensorShape, the underlying
+// storage will only contain active dimensions. In the case of scalar shapes,
+// `NumDims` is allowed to return 0 or 1, but the `storage_` vector will contain
+// 1 element in both cases. In the non-scalar case, `NumDims() ==
+// storage_.size()`.
+class DimsAdapter {
+ public:
+  using StorageType = absl::InlinedVector<int64_t, 4>;
+
+ private:
+  template <typename T>
+  using EnableIfNotTensorShapeType =
+      std::enable_if_t<!std::is_base_of<TensorShapeBase<T>, T>::value>;
+
+  template <typename T>
+  using EnableIfInt = std::enable_if_t<std::is_arithmetic<T>::value &&
+                                       std::is_integral<T>::value>;
+
+  //----- Constructors ------
+
+ public:
+  DimsAdapter() = default;
```

Comment:
I don't see where we use the default constructor. Can you point me to that?

Please also squash your commits if your change is only for addressing this comment.

Reply:
We were using it in TRT_ShapedWeights default constructor. I made it explicit and removed this default constructor. 

Commits squashed.